src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/JsonPayloadGeneratorAdapterUIHandler.java

```diff
@@ -0,0 +1,233 @@
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.zaproxy.zap.extension.fuzz.payloads.DefaultPayload;
+import org.zaproxy.zap.extension.fuzz.payloads.generator.JsonPayloadGenerator;
+import org.zaproxy.zap.extension.fuzz.payloads.generator.PayloadGenerator;
+import org.zaproxy.zap.extension.fuzz.payloads.ui.PayloadGeneratorUI;
+import org.zaproxy.zap.extension.fuzz.payloads.ui.PayloadGeneratorUIHandler;
+import org.zaproxy.zap.extension.fuzz.payloads.ui.PayloadGeneratorUIPanel;
+import org.zaproxy.zap.model.MessageLocation;
+import org.zaproxy.zap.utils.FontUtils;
+import org.zaproxy.zap.utils.ZapNumberSpinner;
+import org.zaproxy.zap.utils.ZapTextArea;
+
+import javax.swing.GroupLayout;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import java.text.MessageFormat;
+
+public class JsonPayloadGeneratorAdapterUIHandler implements
+        PayloadGeneratorUIHandler<DefaultPayload, JsonPayloadGenerator, JsonPayloadGeneratorAdapterUIHandler.JsonPayloadGeneratorUI> {
+    private static final Logger LOGGER = Logger.getLogger(JsonPayloadGeneratorAdapterUIHandler.class);
+    private static final String PAYLOAD_GENERATOR_NAME = getString("name");
+    private static final String PAYLOAD_GENERATOR_DESC = getString("description");
+
+    @Override
+    public String getName() {
+        return PAYLOAD_GENERATOR_NAME;
+    }
+
+    @Override
+    public Class<JsonPayloadGeneratorUI> getPayloadGeneratorUIClass() {
+        return JsonPayloadGeneratorUI.class;
+    }
+
+    @Override
+    public Class<? extends PayloadGeneratorUIPanel<DefaultPayload, JsonPayloadGenerator, JsonPayloadGeneratorUI>> getPayloadGeneratorUIPanelClass() {
+        return JsonPayloadGeneratorUIPanel.class;
+    }
+
+    @Override
+    public PayloadGeneratorUIPanel<DefaultPayload, JsonPayloadGenerator, JsonPayloadGeneratorUI> createPanel() {
+        return new JsonPayloadGeneratorUIPanel();
+    }
+
+    public static class JsonPayloadGeneratorUI implements PayloadGeneratorUI<DefaultPayload, JsonPayloadGenerator> {
+        private final JsonPayloadGenerator jsonPayloadGenerator;
+
+        public JsonPayloadGeneratorUI(JsonPayloadGenerator generator) {
+            this.jsonPayloadGenerator = generator;
+        }
+
+        @Override
+        public Class<? extends JsonPayloadGenerator> getPayloadGeneratorClass() {
+            return JsonPayloadGenerator.class;
+        }
+
+        @Override
+        public String getName() {
+            return PAYLOAD_GENERATOR_NAME;
+        }
+
+        @Override
+        public String getDescription() {
+
+            return MessageFormat.format(
+                    PAYLOAD_GENERATOR_DESC,
+                    jsonPayloadGenerator.getJson());
+        }
+
+        @Override
+        public long getNumberOfPayloads() {
+            try {
+                return jsonPayloadGenerator.getNumberOfPayloads();
+            } catch (Exception e) {
```

Comment:
This can be removed it never happens here.

Reply:
Ouch, I should have been really tired to write something like this :+1: